brion/synapse.cpp

```diff
@@ -45,43 +46,35 @@ namespace
 {
 struct Dataset
 {
-    H5::DataSet dataset;
-    H5::DataSpace dataspace;
-    hsize_t dims[2];
+    std::unique_ptr<HighFive::DataSet> dataset;
+    std::vector<size_t> dims;
```

Comment:
tuple or pair at most if you don't like c-arrays? or you need this for something else?

Reply:
I use this variable as the destination of a HighFive call, if I replace it with a pair I'll have to add more code somewhere else.